src/v/features/feature_table.h

```diff
@@ -510,6 +510,9 @@ class feature_table {
     // applying raft0 log events.
     friend class cluster::feature_backend;
 
+    // for set_applied_offset when applying the bootstrap cmd
+    friend class cluster::bootstrap_backend;
```

Comment:
The pattern used in `bootstrap_original_version()`/`set_original_version()` could be a better fit than befriending yet another class?

Reply:
Agreed, the friend declaration is unfortunate. Though I don't see any benefit in adding a function that does exactly the same as `set_applied_offset`, as an alternative we can make it public, WDYT?

Reply:
The public function may need to add some checks, like a check for the same shardid. If it does not, then just making it public lgtm.